harness/determined/pytorch/_pytorch_context.py

```diff
@@ -92,14 +130,28 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self._reducers = pytorch._PyTorchReducerContext()
         self._determined_profiler = None  # type: Optional[profiler.ProfilerAgent]
 
-        optimizations_config = self.env.experiment_config.get_optimizations_config()
-        self._aggregation_frequency = cast(int, optimizations_config.get("aggregation_frequency"))
-        self._fp16_compression = cast(bool, optimizations_config.get("gradient_compression"))
-        self._average_aggregated_gradients = cast(
-            bool, optimizations_config.get("average_aggregated_gradients")
-        )
-        self._average_training_metrics = cast(
-            bool, optimizations_config.get("average_training_metrics")
+        self._cluster_info = det.get_cluster_info()
```

Comment:
actually, can we just explicitly pass the values that are needed into the PyTorchTrialContext?

This is bad for two reasons.  One: it's reading secret configs other than what is set in `__init__()`.  Two: it's passing a big-blob-of-config into the lowest layers of the training loop.  That is exactly what made the env such a problematic object.  When we went to mock it in unit tests, it was completely unclear which things needed to be mocked and which things could be set with bogus values.

Reply:
good call, and done.